### PR TITLE
docs: add tailwindcss for styles, oidc authorize endpoint playground

### PR DESCRIPTION
**Pull request by peintnermax**
Construct `authorize` endpoint to try out different configurations.

### CODE REVIEW

**Comment by mffap** [comment]
@peintnermax some small observations on first use: 
- I think the button href is not encoded? When I copy the link it works, but seems to have an issue when opening with button (no response type... I guess it escapes after the redirect_uri
- Instance domain trailing slash: we should catch it or describe that we need the trailing slash (easily frustrating)
- the ? after the path should be white

Everything else so far is top!

**Comment by peintnermax** [comment]
> @peintnermax some small observations on first use:
> 
> * I think the button href is not encoded? When I copy the link it works, but seems to have an issue when opening with button (no response type... I guess it escapes after the redirect_uri)
> * Instance domain trailing slash: we should catch it or describe that we need the trailing slash (easily frustrating)
> * the ? after the path should be white
> 
> Everything else so far is top!

had a typo.. reponse_type 😂

File: docs/src/components/authrequest.jsx [hunk 1]
```diff
@@ -0,0 +1,453 @@
...
+        <input
+          className={inputClasses(false)}
+          id="instance"
+          value={instance}
+          onChange={(event) => {
+            const value = event.target.value;
+            setInstance(value);
+          }}
+        />
+        <span className={hintClasses}>The domain of your zitadel instance</span>
```
Comment (mffap) [inline docs/src/components/authrequest.jsx:104]:
```suggestion
        <span className={hintClasses}>The domain of your zitadel instance with trailing slash.</span>
```

File: docs/src/components/authrequest.jsx [hunk 2]
```diff
@@ -0,0 +1,453 @@
...
+                <span className="pointer-events-none absolute inset-y-0 right-0 flex items-center pr-2">
+                  <ChevronUpDownIcon
+                    className="h-5 w-5 text-gray-400"
+                    aria-hidden="true"
+                  />
+                </span>
+              </Listbox.Button>
+              <span className={`${hintClasses} flex`}>
+                This is the resource id of an application. It's the application
+                where you want your users to login.
```
Comment (mffap) [inline docs/src/components/authrequest.jsx:163]:
```suggestion
                Determines whether a code, id_token token or just id_token will be returned. Most use cases will need code.
```

File: docs/src/components/authrequest.jsx [hunk 3]
```diff
@@ -0,0 +1,453 @@
...
+                  <span className="block truncate">{prompt}</span>
+                  <span className="pointer-events-none absolute inset-y-0 right-0 flex items-center pr-2">
+                    <ChevronUpDownIcon
+                      className="h-5 w-5 text-gray-400"
+                      aria-hidden="true"
+                    />
+                  </span>
+                </Listbox.Button>
+                <span className={`${hintClasses} flex`}>
+                  Define if and how the user should be prompted on login.
```
Comment (mffap) [inline docs/src/components/authrequest.jsx:228]:
```suggestion
                  Define how the user should be prompted on login and register.
```

File: docs/src/components/authrequest.jsx [hunk 4]
```diff
@@ -0,0 +1,453 @@
...
+                  .map((checked, i) => (checked ? allScopes[i] : ""))
+                  .filter((s) => !!s)
+                  .join(" ")
+              );
+            }}
+          />
+          <span className={hintClasses}>
+            When requesting the <span className="text-purple-500">scope</span>{" "}
+            <code>urn:zitadel:iam:org:id:{organizationId}`</code>, ZITADEL will
+            enforce that the user is a member of the selected organization
```
Comment (mffap) [inline docs/src/components/authrequest.jsx:345]:
```suggestion
            Enforce organization policies and user membership by requesting the <span className="text-purple-500">scope</span>{" "}
            <code>urn:zitadel:iam:org:id:{organizationId}</code>
```

File: docs/src/components/authrequest.jsx [hunk 5]
```diff
@@ -0,0 +1,453 @@
...
+          </span>
+        </div>
+      </div>
+
+      <div className="py-4">
+        <p className="text-sm mt-0 mb-0 text-purple-500">Scopes</p>
+        <span className={`${hintClasses} flex mb-2`}>
+          Request additional information about the user with scopes. The claims
+          (results of scopes) will be returned on the userinfo_endpoint or in
+          the token (when configured).
```
Comment (mffap) [inline docs/src/components/authrequest.jsx:355]:
```suggestion
          Request additional information about the user with scopes. The claims will be returned on the userinfo_endpoint or in the token (when configured).
```

File: docs/src/components/authrequest.jsx [hunk 6]
```diff
@@ -0,0 +1,453 @@
...
+              const value = event.target.value;
+              setIdTokenHint(value);
+            }}
+          />
+        </div>
+      </div> */}
+
+      <br />
+
+      <div className="rounded-md bg-gray-700 shadow dark:bg-black/10 p-2 flex flex-col items-center">
```
Comment (mffap) [inline docs/src/components/authrequest.jsx:402]: Should have a header "Your Authorization Request"

File: docs/src/components/authrequest.jsx [hunk 7]
```diff
@@ -0,0 +1,453 @@
...
+        <code className="text-sm w-full mb-4 bg-transparent border-none">
+          <span className="text-yellow-500">{instance}</span>
+          <span className="text-white">oauth/v2/authorize</span>
+          <CodeSnipped cname="text-green-500">{`?client_id=${encodeURIComponent(
```
Comment (mffap) [inline docs/src/components/authrequest.jsx:406]:
```suggestion
          <span className="text-white">oauth/v2/authorize?</span>
          <CodeSnipped cname="text-green-500">{`client_id=${encodeURIComponent(
```

File: docs/src/components/authrequest.jsx [hunk 8]
```diff
@@ -0,0 +1,453 @@
...
+          )}&response_type=${encodeURIComponent(
+            responseType
+          )}&scope=${encodeURIComponent(scope)}${
+            prompt ? `&prompt=${encodeURIComponent(prompt)}` : ""
+          }${
+            loginHint && prompt === "select_account"
+              ? `&login_hint=${encodeURIComponent(loginHint)}`
+              : ""
+          }`}
+        >
```
Comment (mffap) [inline docs/src/components/authrequest.jsx:446]: Add plausible

**Review by mffap — CHANGES_REQUESTED** [review]

File: docs/src/components/authrequest.jsx (hunk 8 above)
Comment (peintnermax) [reply docs/src/components/authrequest.jsx]: done but has to be tested as we have no other window.plausible(...) implemented

**Comment by mffap** [comment]
@peintnermax 

- I've added authentication methods with a code challenge
- Moved the playground to its own page
- Added explainer texts

Could you have a quick check, if that looks ok for you? 

The encoding stuff was a bit tricky:`cboa` is apparently deprecated, but I was not able to use the "new" solution, since it required some kind of plugin where webpack was not happy. I think for the moment it's not especially important. I've added the "new" solution as comments.

**Comment by mffap** [comment]
@fforootd @livio-a could you have a quick look at the explainer, please?
I didn't want to duplicate the contents of the API docs. The texts should be to give someone without much knowledge an easier introduction to the different parameters.

**Comment by peintnermax** [comment]
> @peintnermax
> 
> * I've added authentication methods with a code challenge
> * Moved the playground to its own page
> * Added explainer texts
> 
> Could you have a quick check, if that looks ok for you?
> 
> The encoding stuff was a bit tricky:`cboa` is apparently deprecated, but I was not able to use the "new" solution, since it required some kind of plugin where webpack was not happy. I think for the moment it's not especially important. I've added the "new" solution as comments.



> @peintnermax
> 
> * I've added authentication methods with a code challenge
> * Moved the playground to its own page
> * Added explainer texts
> 
> Could you have a quick check, if that looks ok for you?
> 
> The encoding stuff was a bit tricky:`cboa` is apparently deprecated, but I was not able to use the "new" solution, since it required some kind of plugin where webpack was not happy. I think for the moment it's not especially important. I've added the "new" solution as comments.

I was able to replace the deprecated method by importing from 'buffer'. The rest is done nicely 😎

File: docs/docs/apis/openidoauth/authrequest.mdx [hunk 9]
```diff
@@ -0,0 +1,115 @@
+---
+title: OIDC Authentication Request Playground
+sidebar_label: OIDC Playground
+---
+
+import { SetAuthRequest } from "../../../src/components/authrequest";
+
+The OIDC Playground is for testing OpenID Authentication Requests, giving you more insight how OpenID Connect works and how you can customize ZITADEL behavior with different parameters.
+
+An Open ID Connect (OIDC) [authentication request](https://openid.net/specs/openid-connect-core-1_0.html) is an OAuth 2.0 Authorization Request using additional parameters and scopes to request that the end-user be authenticated by ZITADEL.
```
Comment (fforootd) [inline docs/docs/apis/openidoauth/authrequest.mdx:10]:
```suggestion
An OpenID Connect (OIDC) [authentication request](https://openid.net/specs/openid-connect-core-1_0.html) is an OAuth 2.0 Authorization Request using additional parameters and scopes to request that the end-user be authenticated by ZITADEL.
```

File: docs/docs/apis/openidoauth/authrequest.mdx [hunk 10]
```diff
@@ -0,0 +1,115 @@
...
+
+<SetAuthRequest />
+
+## Why this OIDC playground?
+
+In this version of ZITADEL, human end-users need to authenticate trough a hosted login page.
+Your application should initiate a login by issuing an authentication request and redirecting the user to the login page. You can also customize the behavior of ZITADEL by providing additional parameters and scopes in the request.
```
Comment (fforootd) [inline docs/docs/apis/openidoauth/authrequest.mdx:17]:
```suggestion
Currently ZITADEL requires human users to authenticate trough the hosted login page.
Your application should initiate a login by issuing an authentication request and redirecting the user to the login page. You can customize the behavior of ZITADEL by providing additional parameters and scopes in the request.
```

File: docs/docs/apis/openidoauth/authrequest.mdx [hunk 11]
```diff
@@ -0,0 +1,115 @@
...
+
+This playground should help you to initialially craft an authentication request to explore the behavior of ZITADEL in more detail.
```
Comment (fforootd) [inline docs/docs/apis/openidoauth/authrequest.mdx:19]:
```suggestion
This playground should help you to initially craft an authentication request and to explore the behavior of ZITADEL in more depth.
```

File: docs/docs/apis/openidoauth/authrequest.mdx [hunk 12]
```diff
@@ -0,0 +1,115 @@
...
+<p>
+  <span className="text-blue-500">Redirect URI</span> be one of the
+  pre-configured redirect uris for your application. You must add the redirect
+  uri for your application, else you will receive an error.
+</p>
+
+<p>
+  <span className="text-orange-500">Response Type</span> termines whether a
+  code, id_token token or just id_token will be returned. Most use cases will
+  need code.
```
Comment (fforootd) [inline docs/docs/apis/openidoauth/authrequest.mdx:47]:
```suggestion
  <span className="text-orange-500">Response Type</span> defines whether a
  code, id_token token or just id_token will be returned. Most use cases will
  need code.
```

File: docs/docs/apis/openidoauth/authrequest.mdx [hunk 13]
```diff
@@ -0,0 +1,115 @@
...
+Please refer to the full [guide on user metadata](/docs/guides/manage/customize/branding).
+
+### Get user metadata
+
+Pass the scope `urn:zitadel:iam:user:metadata` to get a user's metadata.
+Please refer to the full [guide on branding](/docs/guides/manage/customize/user-metadata) for further details.
+
+### Access core apis
+
+Calling the [core API](/docs/apis/introduction) with the authenticated user, requires that the project ID is included in the audience claim.
```
Comment (fforootd) [inline docs/docs/apis/openidoauth/authrequest.mdx:107]:
```suggestion
Calling the [core API](/docs/apis/introduction) with the authenticated user, requires that the projectID of ZITADEL is included in the audience claim.
```

File: docs/docs/apis/openidoauth/authrequest.mdx [hunk 14]
```diff
@@ -0,0 +1,115 @@
...
+
+This can be achieved by adding the scope `urn:zitadel:iam:org:project:id:zitadel:aud` to your applications authorization request.
+
+## How to use in your project
```
Comment (fforootd) [inline docs/docs/apis/openidoauth/authrequest.mdx:111]:
```suggestion
## How to use ZITADEL in your project
```

**Review by fforootd — COMMENTED** [review]
Some small text improvements and corrections, other than that it looks nice

**Comment by mffap** [comment]
@peintnermax nothing more from my side for the moment. Anything missing from your end? Can you resolve the conflicts? Thanks

**Comment by peintnermax** [comment]
conflicts resolved.. merging is fine from my side 👍

**Review by mffap — APPROVED** [review]
Works like a charm. I think this is good for a first version, we can always improve it later on. Thanks!

**Pull request merged**